cmd/ipfs-cluster-ctl/main.go

```diff
@@ -349,6 +349,10 @@ cluster "pin add".
 					Value: defaultAddParams.ReplicationFactorMax,
 					Usage: "Sets the maximum replication factor for pinning this file",
 				},
+				cli.BoolFlag{
+					Name:  "nocopy",
+					Usage: "Add the URL using filestore. Implies raw-leaves. (experimental)",
```

Comment:
Doesn't it add to the urlstore? I've had some problems figuring out how this is now working in ipfs (I haven' t been able to find the place where the URL is taken from the multipart request and used for the urlstore - I suspect it's somewhere in `unixfs`), but if urlstore is not enabled, adding urls like this on ipfs fails.

Reply:
`WebFile` now implements `FileInfo`, and returns the URL via `FileInfo.AbsPath()`, which is why the upgrade to `go-ipfs-files v0.0.2` is required. This is passed along via the `abspath` header, which ends up in the `ReaderFile`. It is then extracted and set as `DagBuilderParams.fullpath` https://github.com/ipfs/go-unixfs/blob/master/importer/helpers/dagbuilder.go#L85-L88, which in combination with the `nocopy` flag, triggers the same underlying mechanism as `urlstore`, but `urlstore` is not directly used. Only `filestore` is required https://github.com/ipfs/go-ipfs/blob/master/core/coreapi/unixfs.go#L52-L54. Here is the read logic: https://github.com/ipfs/go-ipfs/blob/master/filestore/fsrefstore.go#L138

Reply:
As far as I can tell, enabling `urlstore` makes the command available, but that command just leverages `filestore` internals to actually do the work.

Reply:
Thanks, that's very useful. I have opened https://github.com/ipfs/go-ipfs/issues/6139